include/kernel/faulthandler.h

```diff
@@ -7,11 +7,29 @@
 #pragma once
 
 #include <object.h>
+#ifdef CONFIG_KERNEL_MCS
```

Comment:
can this commit be split into smaller changes?

Reply:
also if its  possible to do this commit without restructuring the code so we can look at the semantics of the change that would help a lot. This commit changes the behaviour and refactors code at the same time. I don't have the state in my head, so the diffs are very hard to understand. 

this looks like it should be implemented in haskell first as well, since it changes the structure of sendIPC in ways that may not work with the haskell spec.

Reply:
Github does not make it clear which commit you are referring to.

Reply:
"mcs: Fault if scheduling unschedulable thread"

yeah I am finding the UI a bit confusing here

Reply:
I think this was a few different commits previously but they were confusing in their own right. I'll see if I can clear up the changes.

Reply:
thanks, this is now much easier to review